src/main/java/uk/gov/register/serialization/RSFExecutor.java

```diff
@@ -70,22 +72,23 @@ private RegisterResult validate(RegisterCommand command, Register register, int
     }
 
     private Boolean validateAppendEntry(RegisterCommand command, Register register, Map<HashValue, Integer> hashRefLine) {
-        String entrySha256 = command.getCommandArguments().get(1);
-        HashValue hashValue = HashValue.decode(HashingAlgorithm.SHA256, entrySha256);
-        if (hashRefLine.containsKey(hashValue)) {
-            hashRefLine.put(hashValue, 0);
-        } else {
-            Optional<Item> item = register.getItemBySha256(hashValue);
-            if (!item.isPresent()) {
-                return true;
+        String delimitedHashes = command.getCommandArguments().get(1);
+        List<HashValue> hashes = Splitter.on(";").splitToList(delimitedHashes).stream()
+                .map(s -> HashValue.decode(SHA256, s)).collect(toList());
+
+        hashes.stream().forEach(hashValue -> {
+            if (hashRefLine.containsKey(hashValue)) {
+                hashRefLine.put(hashValue, 0);
             }
-        }
-        return false;
+        });
+
+        return hashes.stream().allMatch(hashValue -> (!hashRefLine.containsKey(hashValue)
```

Comment:
With derivations, we're also going to get a lot of repeating item-hashes I think. So if we see a item-hash that is not in the RSF, but is in the database, can we store that we've seen it in memory? That way we won't have to go to the database more than necessary. Maybe something like:

```
for (HashValue hashValue : hashes) {
    if (hashRefLine.containsKey(hashValue)) {
        hashRefLine.put(hashValue, 0);
    } else {
        Optional<Item> item = register.getItemBySha256(hashValue);
        if (!item.isPresent()) {
            return true;
        } else {
            hashRefLine.put(hashValue, 0);
        }
    }
}
return false;
```